pkg/apis/core/validation/shoot.go

```diff
@@ -1494,6 +1495,22 @@ func validateMaintenance(maintenance *core.Maintenance, fldPath *field.Path, wor
 	return allErrs
 }
 
+func validateCloudProfile(cloudProfileReference *core.CloudProfileReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if cloudProfileReference == nil {
+		return allErrs
+	}
+	if !sets.New("CloudProfile", "NamespacedCloudProfile").Has(cloudProfileReference.Kind) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("kind"), cloudProfileReference.Kind, "must be CloudProfile or NamespacedCloudProfile"))
+	}
+	if len(cloudProfileReference.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), "must specify a CloudProfile name"))
+	}
+
+	return allErrs
+}
```

Comment:
Could you add unit tests for this?

Reply:
Done in https://github.com/gardener/gardener/pull/9440/commits/026d5bdc62ce06a524e6839f6d663a7dcf36a6e6